### PR TITLE
TOB-AJNA-7: Prevent collision of proposalIds in standard and extraordinary funding

### PR DESCRIPTION
**Pull request by prat-gpt**
* `ProposalId` collision for extraordinary and standard funding is prevented by hashing strings `Extraordinary Funding: ` and `Standard Funding: ` with description hash respectively

# Gas usage
## Pre Change
```| src/grants/GrantFund.sol:GrantFund contract |                 |        |        |        |         |
|---------------------------------------------|-----------------|--------|--------|--------|---------|
| Deployment Cost                             | Deployment Size |        |        |        |         |
| 4657644                                     | 23260           |        |        |        |         |
| Function Name                               | min             | avg    | median | max    | # calls |
| claimDelegateReward                         | 1407            | 57405  | 58540  | 65340  | 193     |
| executeExtraordinary                        | 7152            | 42342  | 39567  | 95588  | 8       |
| executeStandard                             | 8775            | 29860  | 37905  | 47659  | 8       |
| findMechanismOfProposal                     | 613             | 2043   | 799    | 4719   | 3       |
| fundTreasury                                | 8001            | 45104  | 44641  | 65872  | 29      |
| fundingVote                                 | 1584            | 105607 | 105266 | 409345 | 208     |
| getDelegateReward                           | 2975            | 3465   | 2975   | 5017   | 5       |
| getDistributionId                           | 374             | 433    | 374    | 2374   | 502     |
| getDistributionPeriodInfo                   | 1046            | 1955   | 1046   | 5046   | 66      |
| getExtraordinaryProposalInfo                | 992             | 992    | 992    | 992    | 3       |
| getExtraordinaryProposalSucceeded           | 2245            | 2664   | 2740   | 3008   | 3       |
| getFundedProposalSlate                      | 1133            | 1309   | 1368   | 1368   | 4       |
| getFundingPowerVotes                        | 15500           | 15688  | 15751  | 15751  | 4       |
| getFundingVotesCast                         | 1577            | 2317   | 2317   | 3057   | 2       |
| getMinimumThresholdPercentage               | 471             | 1227   | 739    | 2471   | 3       |
| getProposalInfo                             | 980             | 980    | 980    | 980    | 105     |
| getSlateHash                                | 872             | 880    | 884    | 884    | 3       |
| getSliceOfNonTreasury                       | 1577            | 4827   | 4827   | 8077   | 2       |
| getSliceOfTreasury                          | 781             | 1781   | 1781   | 2781   | 2       |
| getTopTenProposals                          | 1187            | 2377   | 2363   | 3304   | 16      |
| getVoterInfo                                | 1002            | 1002   | 1002   | 1002   | 5       |
| getVotesExtraordinary                       | 764             | 7651   | 7681   | 8755   | 449     |
| getVotesFunding                             | 2389            | 6565   | 2797   | 11518  | 15      |
| getVotesScreening                           | 5313            | 5356   | 5347   | 6489   | 401     |
| hasVotedExtraordinary                       | 683             | 1683   | 1683   | 2683   | 2       |
| hashProposal                                | 3843            | 3843   | 3843   | 3843   | 62      |
| proposeExtraordinary                        | 4497            | 59299  | 81926  | 86265  | 15      |
| proposeStandard                             | 4475            | 76991  | 82660  | 82660  | 61      |
| screeningVote                               | 1847            | 42235  | 36714  | 399146 | 227     |
| screeningVotesCast                          | 663             | 1329   | 663    | 2663   | 3       |
| startNewDistributionPeriod                  | 617             | 48521  | 53163  | 75597  | 20      |
| state                                       | 1337            | 4369   | 3842   | 7331   | 18      |
| updateSlate                                 | 971             | 49813  | 69994  | 174519 | 17      |
| voteExtraordinary                           | 565             | 29264  | 29458  | 31424  | 446     |
```

## Post Change
```| src/grants/GrantFund.sol:GrantFund contract |                 |        |        |        |         |
|---------------------------------------------|-----------------|--------|--------|--------|---------|
| Deployment Cost                             | Deployment Size |        |        |        |         |
| 4748368                                     | 23713           |        |        |        |         |
| Function Name                               | min             | avg    | median | max    | # calls |
| claimDelegateReward                         | 1407            | 57127  | 58540  | 65340  | 155     |
| executeExtraordinary                        | 7387            | 43502  | 26445  | 95823  | 6       |
| executeStandard                             | 9009            | 30094  | 38140  | 47894  | 8       |
| findMechanismOfProposal                     | 613             | 2043   | 799    | 4719   | 3       |
| fundTreasury                                | 8001            | 45104  | 44641  | 65872  | 29      |
| fundingVote                                 | 1584            | 105502 | 105113 | 409345 | 170     |
| getDelegateReward                           | 2975            | 3465   | 2975   | 5017   | 5       |
| getDistributionId                           | 374             | 444    | 374    | 2374   | 425     |
| getDistributionPeriodInfo                   | 1046            | 1969   | 1046   | 5046   | 65      |
| getExtraordinaryProposalInfo                | 992             | 992    | 992    | 992    | 3       |
| getExtraordinaryProposalSucceeded           | 2245            | 2664   | 2740   | 3008   | 3       |
| getFundedProposalSlate                      | 1133            | 1309   | 1368   | 1368   | 4       |
| getFundingPowerVotes                        | 15500           | 15688  | 15751  | 15751  | 4       |
| getFundingVotesCast                         | 1577            | 2317   | 2317   | 3057   | 2       |
| getMinimumThresholdPercentage               | 471             | 1227   | 739    | 2471   | 3       |
| getProposalInfo                             | 980             | 980    | 980    | 980    | 104     |
| getSlateHash                                | 872             | 880    | 884    | 884    | 3       |
| getSliceOfNonTreasury                       | 1577            | 4827   | 4827   | 8077   | 2       |
| getSliceOfTreasury                          | 781             | 1781   | 1781   | 2781   | 2       |
| getTopTenProposals                          | 1187            | 2363   | 2363   | 3304   | 16      |
| getVoterInfo                                | 1002            | 1002   | 1002   | 1002   | 5       |
| getVotesExtraordinary                       | 764             | 7531   | 7681   | 8755   | 89      |
| getVotesFunding                             | 2389            | 6565   | 2797   | 11518  | 15      |
| getVotesScreening                           | 5313            | 5359   | 5347   | 6489   | 325     |
| hasVotedExtraordinary                       | 683             | 1683   | 1683   | 2683   | 2       |
| hashProposal                                | 3843            | 3843   | 3843   | 3843   | 59      |
| proposeExtraordinary                        | 4731            | 56052  | 86161  | 86500  | 13      |
| proposeStandard                             | 4709            | 77143  | 82895  | 82895  | 60      |
| screeningVote                               | 1847            | 43022  | 36446  | 399146 | 189     |
| screeningVotesCast                          | 663             | 1329   | 663    | 2663   | 3       |
| startNewDistributionPeriod                  | 617             | 48521  | 53163  | 75597  | 20      |
| state                                       | 1337            | 4162   | 3591   | 7331   | 16      |
| updateSlate                                 | 971             | 48224  | 69994  | 147512 | 17      |
| voteExtraordinary                           | 565             | 28452  | 29458  | 31424  | 86      |

```

### CODE REVIEW

**Review by grandizzy — APPROVED** [review]
LGTM, nice way to fix it!

File: src/grants/base/ExtraordinaryFunding.sol [hunk 1]
```diff
@@ -20,7 +20,12 @@ abstract contract ExtraordinaryFunding is Funding, IExtraordinaryFunding {
     /**
      * @notice The maximum length of a proposal's voting period, in blocks.
      */
-    uint256 internal constant MAX_EFM_PROPOSAL_LENGTH = 216_000; // number of blocks in one month
+    uint256 internal constant MAX_EFM_PROPOSAL_LENGTH                  = 216_000; // number of blocks in one month
```
Comment (MikeHathaway) [inline src/grants/base/ExtraordinaryFunding.sol]: Nit: extra spacing before variable assignment

**Review by MikeHathaway — APPROVED** [review]
Looks great, thanks for the fix. Just had one small nit about spacing

File: src/grants/base/ExtraordinaryFunding.sol (hunk 1 above)
Comment (prat-gpt) [reply src/grants/base/ExtraordinaryFunding.sol]: Fixed in [04c6a7d](https://github.com/ajna-finance/ecosystem-coordination/pull/64/commits/04c6a7d886267d0a9a1efc303bf5d0a295eb1174)

**Pull request merged**